pkg/agent/openflow/multicast_test.go

```diff
@@ -0,0 +1,174 @@
+
+	return externalMcastReceiverFlowBuilder
+}
+
+func newMockMulticastSkipIGMPMetricFlows(ctrl *gomock.Controller) *mocks.MockFlowBuilder {
+	mockFlowBuilder := mocks.NewMockFlowBuilder(ctrl)
+	mockFlowBuilder.EXPECT().Cookie(gomock.Any()).Return(mockFlowBuilder).AnyTimes()
+	mockFlowBuilder.EXPECT().MatchProtocol(gomock.Any()).Return(mockFlowBuilder).AnyTimes()
+	action := mocks.NewMockAction(ctrl)
+	action.EXPECT().NextTable().Return(mockFlowBuilder).AnyTimes()
+	mockFlowBuilder.EXPECT().Action().Return(action).Times(1)
+	mockFlow := mocks.NewMockFlow(ctrl)
+	mockFlowBuilder.EXPECT().Done().Return(mockFlow).Times(1)
+	mockFlow.EXPECT().MatchString().Return("").AnyTimes()
+	return mockFlowBuilder
+}
+
+func newMockIGMPIngressFlowBuilder(ctrl *gomock.Controller) *mocks.MockFlowBuilder {
+	igmpFlowBuilder = mocks.NewMockFlowBuilder(ctrl)
+	igmpFlowBuilder.EXPECT().Cookie(gomock.Any()).Return(igmpFlowBuilder).Times(1)
+	igmpFlowBuilder.EXPECT().MatchProtocol(gomock.Any()).Return(igmpFlowBuilder).Times(1)
+	igmpFlowBuilder.EXPECT().MatchRegMark(gomock.Any()).Return(igmpFlowBuilder).Times(1)
+	action := mocks.NewMockAction(ctrl)
+	action.EXPECT().GotoStage(gomock.Any()).Return(igmpFlowBuilder).Times(1)
+	igmpFlowBuilder.EXPECT().Action().Return(action).Times(1)
+	igmpFlow := mocks.NewMockFlow(ctrl)
+	igmpFlowBuilder.EXPECT().Done().Return(igmpFlow).Times(1)
+	igmpFlow.EXPECT().MatchString().Return("").AnyTimes()
+	return igmpFlowBuilder
+}
+
+func initMockTables(ctrl *gomock.Controller) {
+	mockMulticastEgressRuleTable = mocks.NewMockTable(ctrl)
+	mockMulticastEgressPodMetricTable = mocks.NewMockTable(ctrl)
+	mockMulticastRoutingTable = mocks.NewMockTable(ctrl)
+	mockMulticastOutputTable = mocks.NewMockTable(ctrl)
+	mockMulticastIngressPodMetricTable = mocks.NewMockTable(ctrl)
+
+	bakMulticastEgressRuleTable = MulticastEgressRuleTable.ofTable
+	bakMulticastEgressPodMetricTable = MulticastEgressPodMetricTable.ofTable
+	bakMulticastRoutingTable = MulticastRoutingTable.ofTable
+	bakMulticastOutputTable = MulticastOutputTable.ofTable
+	bakMulticastIngressPodMetricTable = MulticastIngressPodMetricTable.ofTable
+
+	MulticastEgressRuleTable.ofTable = mockMulticastEgressRuleTable
+	MulticastEgressPodMetricTable.ofTable = mockMulticastEgressPodMetricTable
+	MulticastRoutingTable.ofTable = mockMulticastRoutingTable
+	MulticastOutputTable.ofTable = mockMulticastOutputTable
+	MulticastIngressPodMetricTable.ofTable = mockMulticastIngressPodMetricTable
+}
+
+func resetMulticastTables() {
+	MulticastEgressRuleTable.ofTable = bakMulticastEgressRuleTable
+	MulticastEgressPodMetricTable.ofTable = bakMulticastEgressPodMetricTable
+	MulticastRoutingTable.ofTable = bakMulticastRoutingTable
+	MulticastOutputTable.ofTable = bakMulticastOutputTable
+	MulticastIngressPodMetricTable.ofTable = bakMulticastIngressPodMetricTable
+}
+
+func TestInstallMulticastInitialFlows(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	c = prepareClient(ctrl, false)
```

Comment:
I think you can construct your own client focusing on the test itself, but not reuse the one constructed for network policy, e.g.,

```
        bridge := mocks.NewMockBridge(ctrl)
	bridge.EXPECT().AddFlowsInBundle(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
	m := oftest.NewMockOFEntryOperations(ctrl)
	m.EXPECT().AddAll(gomock.Any()).Return(nil).AnyTimes()
	c = &client{
		bridge:      bridge,
		ipProtocols: []binding.Protocol{binding.ProtocolIP},
                cookieAllocator: cookie.NewAllocator(0),
                ofEntryOperations: m,
                eatureMulticast: &mockFeatureMulticast,
                enableMulticast: true,
                enableAntreaPolicy: true,
	}
```

Reply:
done